File: testers/avro/java-async/generated-and-checked-in/com/example/events/consumer/AddressHandler.java
package com.example.events.consumer;

import com.example.events.Address;
import com.example.events.header.StandardHeaders;
import java.util.concurrent.CompletableFuture;

/** Handler interface for address topic events */
public interface AddressHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(String key, Address value, StandardHeaders headers);
}

File: testers/avro/kotlin/generated-and-checked-in/com/example/events/consumer/AddressHandler.kt
package com.example.events.consumer

import com.example.events.Address
import com.example.events.header.StandardHeaders

/** Handler interface for address topic events */
interface AddressHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: Address,
    headers: StandardHeaders
  )
}